### DISCUSSION TITLE
Maya 2022

### DISCUSSION THREAD

**Discussion by wileyknight** [iimachines/Maya2glTF#153] [General]
How can I get this to run for maya 2022?

**Comment by ziriax** [comment]
I didn't find the time to make this work for Maya 2022, but I don't expect many difficulties really...

Are you a developer? If so, did you try building the plugin, and what is failing? If not, I'll try myself as soon as possible😄

**Reply by wileyknight** [reply]
I am a developer and I didn't really give it a look. I just tried to run the batch file and it didn't work on my 2022 so I am using the 2020 version. I'm going to be using this quite a bit in the upcoming weeks so I'll take a look.

Let me know if you can get to it before I do.

Wiley

**Reply by wileyknight** [reply]
From the download zip I went into the package folder and renamed the maya 2020 folder to 2022 and ran the batch. It seems to have put the files in the right locations and when I opened maya I could see the shelf was added and the the icons were there. I went ahead and tried to export a scene I did yesterday from 2020 and I did get an error.

Here is what I got in my maya console:
```
maya2glTF -sn "wileyknight" -of "C:/mydir/" -sf 1 -cpr "(C)2021 wiley"  -sno -vno
maya2glTF: loading plugin...
// Error: file: C:/Users/wiley/Documents/maya/2022/scripts/maya2glTF_UI.mel line 217: Unable to dynamically load : C:/Users/wiley/Documents/maya/2022/plug-ins/maya2glTF.mll
The specified procedure could not be found.
 // 
// Error: file: C:/Users/wiley/Documents/maya/2022/scripts/maya2glTF_UI.mel line 217: The specified procedure could not be found.
 (maya2glTF) // 
```

Like I said the files seem to be in the right directories. So I checked the maya2glTF_UI.mel and the procedure that couldn't be found is error line  217:

`loadPlugin "maya2glTF";`

Thats not much but maybe that'll give you enough to be able to tell me what you think the problem is.

**Comment by ziriax** [comment]
Maya will refuse a plugin that is not specifically build for the corresponding version.

I'll try to install Maya 2022 this week, and see what I can do.

You can't just use the DLLs for another version.

**Reply by ziriax** [reply]
It seems the plugin just builds and works out-of-the-box for Maya2022. Not a single line of code needed to be changed. I'll release it.

**Reply by ziriax** [reply]
See https://github.com/iimachines/Maya2glTF/releases/tag/v1.4.0.

Would you be so kind to test this?

**Reply by wileyknight** [reply]
I ran the batch and exported a scene and everything seems to have exported fine but I did get a warning message for every object exported.

Assign texture coordinates and/or cleanup your mesh and try again please // 
// Warning: line 1: Maya failed to provide the tangents of mesh 'bifrostLiquid1Mesh85Shape'!
Assign texture coordinates and/or cleanup your mesh and try again please // 
// Warning: line 1: Maya failed to provide the tangents of mesh 'bifrostLiquid1Mesh8Shape'!
Assign texture coordinates and/or cleanup your mesh and try again please // 
// Warning: line 1: Maya failed to provide the tangents of mesh 'bifrostLiquid1Mesh9Shape'!

**Reply by ziriax** [reply]
That means that your mesh is missing some information (usually bad or missing UVs or zero area polygons). Maya cannot generate tangents for these.

You can try to export with the `-mts` flag, this bypasses Maya’s tangents, and uses industry standard ones.

Thanks for testing the plug-in!

**Reply by wileyknight** [reply]
The plugin has worked for everything I've needed so far.

Thank you for creating this. It's perfect for exporting models for three.js web use.
